src/commons/eventsourcing/kv.rs

```diff
@@ -91,11 +91,19 @@ impl KeyValueStore {
         let mut base = work_dir.to_path_buf();
         base.push(name_space);
 
-        if !base.exists() {
-            file::create_dir(&base)?;
+        let store = KeyValueStore::Disk(KeyValueStoreDiskImpl { base });
+
+        match &store {
+            KeyValueStore::Disk(disk_store) => {
+                // If this is a new store then initialise the disk and set the version
+                if !disk_store.base.exists() {
+                    file::create_dir(&disk_store.base)?;
```

Comment:
Should this be `create_dir_all()`?

Reply:
`file::create_dir` uses `fs::create_dir_all` under the hood. It could be renamed I suppose